packages/vscode-extension/src/debug/teamsfxTaskProvider.ts

```diff
@@ -151,7 +152,31 @@ export class TeamsfxTaskProvider implements vscode.TaskProvider {
     task: vscode.Task,
     token?: vscode.CancellationToken | undefined
   ): Promise<vscode.Task | undefined> {
-    // Return undefined since all tasks are provided and fully resolved
+    if (task.definition.type !== TeamsfxTaskProvider.type || !task.definition.command) {
```

Comment:
Why implement in `resolveTask` rather than `provideTasks`?

Reply:
As the discussion [here](https://github.com/microsoft/vscode/issues/58836)
> As the extension author you can implement resolveTask to allow your extension to handle any tasks of your task type that appear in the user's tasks.json but aren't returned by provideTasks. I assume your provideTasks doesn't return a task with your specific args value, so the VS Code tasks system has no idea what to do with it. If you implement resolveTask, you can provide a Task for tasks.json tasks that your provideTasks doesn't return.
> Looks like the documentation doesn't outline this well. I'll update it.

Reply:
I believe the key point is:
- If using `provideTasks` and user modifies `args` before F5, will the updated `args` be passed to vsc?

Reply:
Yes, I can't list all args combinations in the `provideTasks`.

Reply:
No need to list all args combinations, but just `args` itself. `args` object will be handled by custom execution dynamically.

BTW, one possible reason to use `resolve` instead of `provide` is - we may need to dynamically adjust `presentation` or `runOptions` per `args`.

Reply:
I see. `provideTasks` cannot be customized.